controllers/awscluster_controller_test.go

```diff
@@ -13,42 +13,1154 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	"sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/services"
+	ec2Service "sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/services/ec2"
+	"sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/services/ec2/mock_ec2iface"
+	elbService "sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/services/elb"
+	"sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/services/elb/mock_elbiface"
+	"sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/services/network"
+	"sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/services/securitygroup"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/util"
 )
 
-func TestAWSClusterReconciler(t *testing.T) {
-	g := NewWithT(t)
-	ctx := context.Background()
+func TestAWSClusterReconciler_IntegrationTests(t *testing.T) {
+	var (
+		reconciler AWSClusterReconciler
+		mockCtrl   *gomock.Controller
+		recorder   *record.FakeRecorder
+	)
+	var secret = &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-secret",
+			Namespace: "capa-system",
+		},
+		Data: map[string][]byte{
+			"AccessKeyID":     []byte("access-key-id"),
+			"SecretAccessKey": []byte("secret-access-key"),
+			"SessionToken":    []byte("session-token"),
+		},
+	}
+
+	setup := func(t *testing.T, g *WithT) {
+		t.Helper()
+		mockCtrl = gomock.NewController(t)
+		recorder = record.NewFakeRecorder(10)
+		reconciler = AWSClusterReconciler{
+			Client:   testEnv.Client,
+			Recorder: recorder,
+		}
+	}
 
-	reconciler := &AWSClusterReconciler{
-		Client: testEnv.Client,
+	teardown := func(t *testing.T, g *WithT) {
+		t.Helper()
+		mockCtrl.Finish()
 	}
 
-	instance := &infrav1.AWSCluster{ObjectMeta: metav1.ObjectMeta{Name: "foo", Namespace: "default"}}
-	instance.Default()
+	t.Run("Should not create VPC in case of unmanaged VPC", func(t *testing.T) {
+		g := NewWithT(t)
+		mockCtrl = gomock.NewController(t)
+		ec2Mock := mock_ec2iface.NewMockEC2API(mockCtrl)
+		elbMock := mock_elbiface.NewMockELBAPI(mockCtrl)
+		expect := func(m *mock_ec2iface.MockEC2APIMockRecorder, e *mock_elbiface.MockELBAPIMockRecorder) {
+			mockedCreateVPCCalls(m)
+			mockedCreateSGCalls(m)
+			mockedCreateLBCalls(e)
+			mockedDescribeInstanceCall(m)
+		}
+		expect(ec2Mock.EXPECT(), elbMock.EXPECT())
+
+		capaNS := &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "capa-system",
+			},
+		}
+		g.Expect(testEnv.Create(ctx, capaNS)).To(Succeed())
+
+		controllerIdentity := createControllerIdentity(g)
+		ns, err := testEnv.CreateNamespace(ctx, fmt.Sprintf("integ-test-%s", util.RandomString(5)))
+		g.Expect(err).To(BeNil())
+		setup(t, g)
+		awsCluster := getAWSCluster("test", ns.Name)
+		g.Expect(reconciler.Client.Create(ctx, secret)).To(Succeed())
+
+		g.Expect(testEnv.Create(ctx, &awsCluster)).To(Succeed())
+		defer teardown(t, g)
+		defer t.Cleanup(func() {
+			g.Expect(testEnv.Cleanup(ctx, controllerIdentity)).To(Succeed())
+			g.Expect(testEnv.Cleanup(ctx, &awsCluster)).To(Succeed())
+		})
+
+		cs, err := scope.NewClusterScope(
```

Comment:
We can refactor NewClusterScope into a method as this piece is repeated in all tests.